src/generic_solver.cpp

```diff
@@ -357,7 +382,18 @@ std::string GenericSolver::to_smtlib_def(Term term) const
         result += " " + (*term_name_map)[c];
       }
     }
-    result += ")";
+    if (gt->get_op() == Apply_Constructor)
+    {
+      shared_ptr<GenericDatatype> dt = static_pointer_cast<GenericDatatype>(
+          (gt->get_sort())->get_datatype());
+      result += dt->get_num_selectors((*term_name_map)[gt->get_children()[0]])
+                    ? ")"
+                    : "";
```

Comment:
Instead of repeating the check here, it would be a bit better to define some boolean, e.g., `nullary_constructor` and only perform the check once.